nixos/modules/system/boot/systemd/initrd.nix

```diff
@@ -333,9 +333,12 @@ in {
     system.build = { inherit initialRamdisk; };
 
     boot.initrd.availableKernelModules = [
-      "autofs4"           # systemd needs this for some features
-      "tpm-tis" "tpm-crb" # systemd-cryptenroll
-    ];
+      "autofs4" # systemd needs this for some features
+    ]
+      # systemd-cryptenroll
+      ++ lib.optional (config.boot.kernelPackages.kernel.config.isEnabled "TCG_TIS") "tpm-tis"
+      ++ lib.optional (config.boot.kernelPackages.kernel.config.isEnabled "TCG_CRB") "tpm-crb"
```

Comment:
Does this actually evaluate to true on our default kernel config?

Also, for the record, I solved this issue downstream by applying the following overlay:
```nix
(self: super: {
  makeModulesClosure = { kernel, firmware, rootModules, allowMissing ? true }:
    super.callPackage "${super.path}/pkgs/build-support/kernel/modules-closure.nix" {
      inherit kernel firmware rootModules;
      allowMissing = true;
    };
})
```
Not having that sanity check sucks, but at least it makes my systems build and they work just fine.

Reply:
So this would end up disabling these by default. Is that what we want?

Reply:
We probably want to enable *them* explicitly in our kernel configuration, right? cc @K900 who might have an idea

Reply:
> Does this actually evaluate to true on our default kernel config?

No, despite `TCG_TIS` `TCG_CRB` are `m` for my default setting. The condition here doesn't work as expected.

The default kernel config cannot be accessed during evaluation, and we don't have a mechanism to check like that. `pkgs.linuxPackages.kernel.config` contains only `CONFIG_FW_LOADER`, `CONFIG_MODULES` and nothing else for me.

As a reference, `system.requiredKernelConfig` simply skip all checks for the default kernel.
https://github.com/NixOS/nixpkgs/blob/a575c243c23e2851b78c00e9fa245232926ec32f/nixos/modules/system/boot/kernel.nix#L345-L349

I propose to add a new option to guard the initrd TPM2 feature, which then sets `system.requiredKernelConfig` and `boot.initrd.availableKernelModules`. So it enables by default, and verifies for customized kernel.